### PR TITLE
Add staking accounts to the membership pallet.

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
### Changes
- add `confirm_staking_account` extrinsic
- add `add_staking_account_candidate` extrinsic
- add `remove_staking_account` extrinsic
- add `StakingAccountValidator` trait to the common module
- implement `StakingAccountValidator` in the membership module
- refactor staking handler and working group

[Main issue](https://github.com/Joystream/joystream/issues/1736)

### CODE REVIEW

File: runtime-modules/membership/src/lib.rs [hunk 1]
```diff
@@ -684,6 +741,88 @@ decl_module! {
...
+        pub fn confirm_staking_account(
+            origin,
+            member_id: T::MemberId,
+        ) {
+            let staking_account_id = ensure_signed(origin)?;
+
+            let membership = Self::ensure_membership(member_id)?;
+
+            ensure!(
+                membership.staking_account_exists(&staking_account_id),
```
Comment (bedeho) [inline runtime-modules/membership/src/lib.rs]: Should we not check that its not confirmed already?

**Review by bedeho — COMMENTED** [review]
Currently a single staking account can be associated with an unlimited number of memberships. While in principle it _may_ be that this still is safe, as long as all staking on such accounts uses `is_account_free_of_conflicting_stakes` implementation that rejects reuse a lock type with itself, but I am still not 100% sure about that, and also its not the mental model we want to encourage. Having a storage map from equivalent to `account => member_idXstatus` would solve the problem, and also remove need for any staking account cap.

What do you think?

File: runtime-modules/membership/src/lib.rs (hunk 1 above)
Comment (shamil-gadelshin) [reply runtime-modules/membership/src/lib.rs]: Ok. I added this check.

**Comment by shamil-gadelshin** [comment]
I changed staking account verification to the double map. I removed the limit for the staking account number and added "staking account could be used only once in the system" check.

**Comment by shamil-gadelshin** [comment]
Review fixes:
- changed double map to map for the staking account registration
- changed the workflow (adding and removing account requires staking account origin, confirmation - member origin).

**Review by bedeho — APPROVED** [review]
Excellent work.

**Pull request merged**